todo_app/app.py

```diff
@@ -71,51 +21,41 @@ def contact():
     @app.route('/', methods=['GET'])
     def get_index():
         cardslist = []
-        for status_mapping in statusMappingList:
-            cbl_board = TrelloClient()
-            result = cbl_board.get_ListCards(status_mapping.list_id)    
-            if result.status_code == 200:
-                for card in result.json():
-                    if not (card['due'] is None):
-                        card['due'] = (card['due']).split("T")[0]
-                    card['dateLastActivity'] = (card['dateLastActivity']).split("T")[0]
-                    cardslist.append( Card(card=card, statuslabel=status_mapping.status) )    
-            else:
-                return render_template("error.html",error="failed to get Trello cards!")    
+        mongo = MongoDBClient()
+        items = mongo.get_AllItems()
+
+        for item in items:
+            cardslist.append(Card(item))
+
         item_view_model = ViewModel(cardslist)
         return render_template('index.html', view_model=item_view_model)
 
     @app.route('/getpreviousdonetasks', methods=['GET'])
     def get_previous_done_tasks():
         cardslist = []
-        for status_mapping in statusMappingList:
-            cbl_board = TrelloClient()
-            result = cbl_board.get_ListCards(status_mapping.list_id)    
-            if result.status_code == 200:
-                for card in result.json():
-                    if not (card['due'] is None):
-                        card['due'] = (card['due']).split("T")[0]
-                    card['dateLastActivity'] = (card['dateLastActivity']).split("T")[0]
-                    cardslist.append( Card(card=card, statuslabel=status_mapping.status) )    
-            else:
-                return render_template("error.html",error="failed to get Trello cards!")    
+        mongo = MongoDBClient()
+        qry = {
+            "status": "Done",
+            "dateLastActivity": {"$lt": datetime.datetime.strptime((datetime.date.today()).strftime("%Y-%m-%d"), '%Y-%m-%d')}    
+        }
+        items = mongo.get_qryItems(qry)
```

Comment:
Just from a code cleanliness perspective, it might be nice to have these queries within your MongoDbClient as methods; so here you could just call `mongo.get_older_done_items()` or something, and encapsulate the query details

Reply:
TY. Updated this as part of Mod 12.